pkg/controller/gitopsconfig/jobmonitor.go

```diff
@@ -0,0 +1,237 @@
+package gitopsconfig
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	gitopsv1alpha1 "github.com/KohlsTechnology/eunomia/pkg/apis/eunomia/v1alpha1"
+	"gopkg.in/robfig/cron.v2"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type jobmonitor struct {
+	client       client.Client
+	Name         string
+	Namespace    string
+	NextSchedule time.Time
+}
+
+//constant instead of using string
+const (
+	inprogress = "INPROGRESS ..."
+	success    = "SUCCESS ..."
+	failed     = "FAILED ..."
+)
+
+//sync.Map is used inorder to introduce syncronization for reading and writing the data
+var runningJobMap sync.Map
+
+func scheduleStatusForCronJobs(job jobmonitor, schedule string, instanceName string, instanceNamespace string) {
+	result, err := job.getCronjobmonitor()
+
+	if err != nil {
+		log.Error(err, "Error in GetCronJob")
+		return
+	}
+
+	//If reconciler calls the same cronjob before the final update of status, we might have an error while
+	//updating the status. To avoid this error runningJobMap is used.
```

Comment:
Thank you *very* much for adding those kinds of good comments, explaining the rationale info for *why* some things were done, which is often not easy or even impossible to otherwise find in code. **Highly appreciated!**